master/reference/advanced/etcd-rbac/kubernetes.md

```diff
@@ -0,0 +1,204 @@
+created for the specific component with the information above.
+
+## Kubernetes API server
+
+The kube-apiserver is the only Kubernetes component that directly accesses etcd.
+The flags required to provide the kube-apiserver with certificates for
+accessing an etcd cluster are:
+- `--etcd-cafile=<CA certificate`
+- `--etcd-certfile=<certificate with etcd username as CN>`
+- `--etcd-keyfile=<key for the above certificate>`
+
+## Calico components
+
+### Single cert setup
+
+The simplest setup is to use Calico with a single certificate that is tied to
+roles for accessing all the etcd paths needed.  The downside to this is that
+all Calico components have the same access rights meaning some components could
+access paths that they do not need.
+With the one certificate and key,
+[this manifest]({{site.baseurl}}/{{page.version}}/getting-started/kubernetes/installation/hosted/calico.yaml)
+can be updated with the CA, certificate, and key information.
+
+The pieces that would need updating are:
+- The `calico-config` ConfigMap
+    - The line `etcd_ca: ""   # "/calico-secrets/etcd-ca"` should be updated
+	  to `etcd_ca: "/calico-secrets/etcd-ca"`.
+	- The line `etcd_cert: "" # "/calico-secrets/etcd-cert"` should be updated
+	  to `etcd_cert: "/calico-secrets/etcd-cert"`.
+	- The line `etcd_key: ""  # "/calico-secrets/etcd-key"` should be updated
+	  to `etcd_key: "/calico-secrets/etcd-key"`.
+- The Secret named `calico-etcd-secrets` needs to be updated with the CA and
+  cert/key. The information stored in data in a Secret needs to be base64
+  encoded.  The files can be converted to base64 encoding by doing a command
+  like `cat <file> | base64 -w 0` on each file and then inserting the output
+  to the appropriate field.
+    - The `etcd-key` field needs the base64 encoded file contents from the
+	  key file.
+	- The `etcd-cert` field needs the base64 encoded file contents from the
+	  certificate file.
+	- The `etcd-ca` field needs the base64 encoded file contents from the
+	  Certificate Authority certificate.
+
+### Multiple cert setup
+
+A slightly more complicated setup that needs a certificate for each component
+is possible while still using a hosted manifest.  This setup requires a
+certifcate for each different Calico component type listed above (cni-plugin,
+Calico policy controller, and calico/node).
+
+With this setup the `calico-config` ConfigMap needs the same updates as were
+made in the [Single cert setup](#single-cert-setup).  The difference in this
+setup is a separate Secret for each component must be created with the CA,
+certificate, and key data base64 encoded.  Then the Secret for each component
+must be in the `volumes` list for each pod and the `volumeMounts` for a
+container referencing the volume for the `/calico-secrets` mountPath.
+
+The example below is to help clarify the Secrets that are needed and show
+how they are associated with the proper pod and container within the pod.
+
+```
```

Comment:
Oh, actually this `Data omitted` stuff.  Is there anything we can do to make it a little more - run this command and K8s is configured?

Let's chat about this as I haven't fully grok'd it yet anyhoooo.